File: go-version/leetcode/LC2348/zero_filled_subarray.go
package LC2348

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/8/19 下午11:26
 * @FilePath: zero_filled_subarray
 * @Description:
 */

func zeroFilledSubarray(nums []int) int64 {
	var cnt, ans int64
	for _, num := range nums {
		if num == 0 {
			cnt++
			ans += cnt
		} else {
			cnt = 0
		}
	}
	return ans
}

File: python-version/LC2348/zero_filled_subarray.py
from typing import List


class Solution:
    @staticmethod
    def zeroFilledSubarray(nums: List[int]) -> int:
        cnt, ans = 0, 0
        for num in nums:
            if num == 0:
                cnt += 1
                ans += cnt
            else:
                cnt = 0
        return ans
